### DISCUSSION TITLE
help on how to use useAxios i get undefined when i try to access the returned data

### DISCUSSION THREAD

**Discussion by thamibn** [vueuse/vueuse#769] [Q&A] [answered]
when i use useAxios(url) when i console the data it shows undefined, i am not sure what i am doing wrong or i am missing.
can someone please help
```
setup(){
        const { data, isFinished } = useAxios('users/json')
        console.log(data) //this consoles refImp...
        console.log(data.value) // this consoles undefined
    },
```
![Screenshot from 2021-09-18 19-14-32](https://user-images.githubusercontent.com/28577943/133897017-ef52f21c-8be7-4829-bbff-f152ea2e2151.png)

**Comment by thamibn** [comment]
ive also tried aysnc await

```
const test = ref(null)
        onMounted(async () => {
            const { data, isFinished } = await useAxios('users/json');
            test.value = data.value
        });
        console.log(test)
        ```

**Comment by sibbng** [comment, marked as answer]
Because request is not finished yet. `data` is reactive value and will be updated as soon as request finished, and it will render your template if you used it on your template. 

If you want to log data result on script block try watch.

```ts
watch(data, () => {
  console.log(data.value)
})

```

Refs Docs: https://v3.vuejs.org/api/refs-api.html#refs
Watch Docs: https://v3.vuejs.org/guide/reactivity-computed-watchers.html#watch

**Reply by thamibn** [reply]
So I cannot await it inside the unmounted?
